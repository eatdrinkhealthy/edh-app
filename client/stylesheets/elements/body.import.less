body {
   font-family: "Roboto", sans-serif;
   background-color: white;
   height: 100%;
   width: 100%;
   overflow-x: hidden;
   &.unselectable {
      -webkit-touch-callout: none;
      -webkit-user-select: none;
      user-select: none;
      -webkit-tap-highlight-color: rgba(0,0,0,0);
      -webkit-tap-highlight-color: transparent;
   }
}
