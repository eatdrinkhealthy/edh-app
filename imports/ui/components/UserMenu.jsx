// @flow
import React, { Component } from "react";
import { Link } from "react-router-dom";

class UserMenu extends Component {
  props: {
    username: ?string,
    logout: () => void,
  };

  renderLoggedOut(): React$Element<*> {  // eslint-disable-line class-methods-use-this
    return (
      <span>
        <Link
          id="joinLink"
          className="btn btn-default"
          to={{ pathname: "/sidebar", search: "?action=signup" }}
        >
          JOIN
        </Link>
        <Link
          id="loginLink"
          className="btn btn-default"
          to={{ pathname: "/sidebar", search: "?action=login" }}
        >
          LOG IN
        </Link>
      </span>
    );
  }

  renderLoggedIn = (): React$Element<*> => (
    <span>
      <span id="loggedInUser" className="">{this.props.username}</span>
      <a
        id="logoutLink"
        className="btn btn-default"
        onClick={this.props.logout}
      >
        Logout
      </a>
    </span>
  );

  render() {  // eslint-disable-line flowtype/require-return-type
    return this.props.username ? this.renderLoggedIn() : this.renderLoggedOut();
  }
}

export default UserMenu;
