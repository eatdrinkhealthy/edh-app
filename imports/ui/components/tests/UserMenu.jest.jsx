// @flow
/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import UserMenu from "../UserMenu";
import { elements as es } from "../../../../tests/end-to-end/elements";

describe("<UserMenu />", function () {
  it("matches render snapshot - null username", function () {
    const tree = renderer.create(
      <MemoryRouter>
        <UserMenu username={null} logout={jest.fn()} />
      </MemoryRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("matches render snapshot - with username", function () {
    const tree = renderer.create(
      <MemoryRouter>
        <UserMenu username="testUser" logout={jest.fn()} />
      </MemoryRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should display the username when passed a user name", function () {
    const wrapper = shallow(<UserMenu username="testUser" logout={jest.fn()} />);
    expect(wrapper.find(es.userMenu.username).text()).toBe("Welcome, testUser!");
  });

  it("should not show login & join buttons when logged in, and show logout", function () {
    const wrapper = shallow(<UserMenu username="testUser" logout={jest.fn()} />);
    expect(wrapper.find(es.userMenu.joinLink).length).toBe(0);
    expect(wrapper.find(es.userMenu.loginLink).length).toBe(0);
    expect(wrapper.find(es.userMenu.logoutLink).length).toBe(1);
  });

  it("should call the logout prop function when clicking the logout button", function () {
    const logoutFn = jest.fn();

    const wrapper = shallow(<UserMenu username={"testUser"} logout={logoutFn} />);
    wrapper.find("a").simulate("click");
    expect(logoutFn).toHaveBeenCalled();
  });
});
