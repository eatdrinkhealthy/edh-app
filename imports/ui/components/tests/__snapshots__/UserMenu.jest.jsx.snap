// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserMenu /> matches render snapshot - null username 1`] = `
<span>
  <a
    className="dib f5 link dim ma1 mb2 ph3 pv2 ba bw2 br3 white bg-purple b--purple"
    href="/sidebar"
    id="joinLink"
    onClick={[Function]}
  >
    JOIN
  </a>
  <a
    className="dib f5 link dim ma1 mb2 ph3 pv2 ba bw2 br3 b--purple"
    href="/sidebar"
    id="loginLink"
    onClick={[Function]}
  >
    LOGIN
  </a>
</span>
`;

exports[`<UserMenu /> matches render snapshot - with username 1`] = `
<span>
  <span
    className="f5"
    id="loggedInUser"
  >
    testUser
  </span>
  <a
    className="dib f5 link dim ma1 mb2 ph3 pv2 ba bw2 br3 white bg-purple b--purple"
    id="logoutLink"
    onClick={[Function]}
  >
    Logout
  </a>
</span>
`;
